contexts/uuid/uuid.go

```diff
@@ -0,0 +1,45 @@
+// Package uuid provides helpers for propagating UUIDs accross services.
+package uuid
+
+import (
+	"context"
+
+	grpctransport "github.com/go-kit/kit/transport/grpc"
+	"github.com/google/uuid"
+	"google.golang.org/grpc/metadata"
+)
+
+// Use a private type to prevent name collisions with other packages.
+type key string
+
+const uuidKey key = "UUID"
+
+// NewContext creates a new context with the UUID set to the provided value
+func NewContext(ctx context.Context, uuid string) context.Context {
+	return context.WithValue(ctx, uuidKey, uuid)
+}
+
+// FromContext returns the UUID value stored in ctx, if any.
+func FromContext(ctx context.Context) (string, bool) {
+	uuid, ok := ctx.Value(uuidKey).(string)
+	return uuid, ok
+}
+
+// NewRandom returns a Random (Version 4) UUID string.
+func New() string {
+	uuid, err := uuid.NewRandom()
+	if err != nil {
+		return ""
```

Comment:
It seems a little concerning to me that we return an empty  string here and don't propagate the error in any way.

Reply:
We could add a panic but IMO that would be worse. The matter of fact is that the request ID is not critical. If a request gets an empty string assigned it shouldn't break our API's. 

OTOH I would not use this helper for creating UUIDs for other tasks. It's intended only for propagation. Maybe the documentation or the name should reflect that? `uuid.PropagationUUID()` ?

Reply:
These UUIDs are intended for monitoring across services? In that case I think it's okay, but maybe we can at least add a comment suggesting not to use this function for other purposes?